### PR TITLE
DCS-1348 Adding integration tests

### PR DESCRIPTION
**Pull request by andrewrlee**
added eslint plugin to prevent `it.only` from being committed

### CODE REVIEW

File: integration_tests/.eslintrc [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
-  "plugins": ["cypress"],
+  "plugins": ["cypress", "no-only-tests"],
```
Comment (andrewrlee) [inline integration_tests/.eslintrc:2]: Prevent `it.only` from being committed

File: integration_tests/integration/bookedtoday/arrivals/existingRecord.spec.ts [hunk 2]
```diff
@@ -18,31 +18,31 @@ context('Confirm Arrival', () => {
   it('PER record should contain a full set of correctly formatted move data', () => {
     cy.task('stubExpectedArrival', expectedArrival)
     cy.signIn()
-    const confirmArrivalPage = ConfirmArrivalPage.goTo(expectedArrival.id)
+    const existingRecordPage = ExistingRecordPage.goTo(expectedArrival.id)
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/arrivals/existingRecord.spec.ts:21]: split this ConfirmArrivalPage into two - `ExistingRecordPage` and `NoExistingRecordPage`. 


Trying to model as one page was a bit messy

File: integration_tests/integration/bookedtoday/arrivals/sex.spec.ts [hunk 3]
```diff
@@ -26,8 +26,6 @@ context('Sex', () => {
     cy.signIn()
     const sexPage = SexPage.goTo(expectedArrival.id)
     sexPage.continue().click()
-    sexPage.errorSummaryTitle().contains('There is a problem')
-    sexPage.errorSummaryBody().contains('Select a sex')
-    sexPage.errorSummaryMessage().contains('Select a sex')
+    sexPage.hasError('Select a sex')
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/arrivals/sex.spec.ts:29]: extracted out helper method

File: integration_tests/integration/bookedtoday/choosePrisoner.spec.ts [hunk 4]
```diff
@@ -28,46 +31,33 @@ context('Choose Prisoner', () => {
...
-    choosePrisonerPage.expectedArrivalsFromCustodySuite(1).should('contain.text', 'Prisoner, Mark')
-    choosePrisonerPage.expectedArrivalsFromCustodySuite(2).should('contain.text', 'Smith, Barry')
-    choosePrisonerPage.expectedArrivalsFromCustodySuite(3).should('contain.text', 'Smith, Bob')
-
-    choosePrisonerPage.expectedArrivalsFromAnotherEstablishment(1).should('contain.text', 'Offender, Karl')
-  })
 
-  it('Should handle prison numbers and PNC numbers both present and unavailable', () => {
-    cy.signIn()
-    const choosePrisonerPage = ChoosePrisonerPage.goTo()
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/choosePrisoner.spec.ts:46]: merged these two tests

File: integration_tests/integration/bookedtoday/choosePrisoner.spec.ts [hunk 5]
```diff
@@ -131,32 +121,93 @@ context('Choose Prisoner', () => {
...
-    canStartToConfirmArrival(choosePrisonerPage, expectedArrivals.court.notMatched, 'COURT', 3)
+    const choosePrisonerPage = ChoosePrisonerPage.goTo()
+    choosePrisonerPage.arrivalFrom('COURT')(1).confirm().click()
 
-    canStartToConfirmArrival(choosePrisonerPage, expectedArrivals.custodySuite.notCurrent, 'CUSTODY_SUITE', 2)
-    canStartToConfirmArrival(choosePrisonerPage, expectedArrivals.custodySuite.notMatched, 'CUSTODY_SUITE', 3)
+    Page.verifyOnPage(NoExistingRecordPage)
   })
 
-  it.only('Only court arrivals with a current booking will have a link leading to the check court return page', () => {
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/choosePrisoner.spec.ts:148]: `it.only` meant some of these tests were actually broken!

File: integration_tests/integration/bookedtoday/choosePrisoner.spec.ts [hunk 6]
```diff
@@ -131,32 +121,93 @@ context('Choose Prisoner', () => {
       })
   })
 
-  it('Only court arrivals without a current booking, arrivals from custody suites and prison transfers will have a link leading to the Confirm arrival page', () => {
+  it('Current bookings from court are processed as court returns', () => {
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/choosePrisoner.spec.ts:124]: Having tests for each of our main stories

File: integration_tests/integration/bookedtoday/choosePrisoner.spec.ts [hunk 7]
```diff
@@ -172,18 +223,4 @@ context('Choose Prisoner', () => {
...
     choosePrisonerPage.arrivalFrom('COURT')(3).confirm().should('not.exist')
   })
-
-  function canStartToConfirmArrival(
-    choosePrisonerPage: ChoosePrisonerPage,
-    arrival: Record<string, string | boolean | unknown[]>,
-    arrivalType: 'COURT' | 'PRISON' | 'CUSTODY_SUITE',
-    rowNumber: number
-  ) {
-    cy.task('stubExpectedArrival', arrival)
```
Comment (andrewrlee) [inline integration_tests/integration/bookedtoday/choosePrisoner.spec.ts:182]: This was failing as we now have new pages to go to

File: integration_tests/integration/featureNotAvailable.spec.ts [hunk 8]
```diff
@@ -19,17 +18,17 @@ context('Feature not available', () => {
...
+      potentialMatches: [expectedArrivals.potentialMatch],
+    })
     cy.task('stubExpectedArrival', expectedArrival)
     cy.task('stubCreateOffenderRecordAndBookingReturnsError', { arrivalId: expectedArrival.id, status: 400 })
 
     cy.signIn()
-    const confirmArrivalPage = ConfirmArrivalPage.goTo(expectedArrival.id)
-    confirmArrivalPage.continue().click()
 
-    const sexPage = Page.verifyOnPage(SexPage)
```
Comment (andrewrlee) [inline integration_tests/integration/featureNotAvailable.spec.ts:30]: removing reference to unrelated feature

File: integration_tests/integration/temporaryabsences/temporaryAbsences.spec.ts [hunk 9]
```diff
@@ -76,7 +76,7 @@ context('A user can view all current temporary absences', () => {
     temporaryAbsencesPage.temporaryAbsences(4).confirm().should('not.exist')
   })
 
-  it.only('Links shown for a reception user navigate to confirm a returning prisoner page', () => {
```
Comment (andrewrlee) [inline integration_tests/integration/temporaryabsences/temporaryAbsences.spec.ts:79]: `it.only`! Thankfully these tests were still passing!

File: integration_tests/mockApis/responses/expectedArrivals.ts [hunk 10]
```diff
@@ -1,4 +1,32 @@
 export default {
+  arrival: ({
+    fromLocationType,
+    isCurrentPrisoner,
+    dateOfBirth = '1970-01-01',
+    pncNumber = '01/2345A',
+    prisonNumber = 'G0015GF',
+    potentialMatches = [],
+  }) => ({
```
Comment (andrewrlee) [inline integration_tests/mockApis/responses/expectedArrivals.ts:9]: helper method to allow creating arrivals in a way which emphasises the salient aspects required for the specific test

File: integration_tests/pages/bookedtoday/arrivals/imprisonmentStatus.ts [hunk 11]
```diff
@@ -10,12 +10,6 @@ export default class ImprisonmentStatusPage extends Page {
     return Page.verifyOnPage(ImprisonmentStatusPage)
   }
 
-  errorSummaryTitle = (): PageElement => cy.get('#error-summary-title')
-
-  errorSummaryBody = (): PageElement => cy.get('.govuk-error-summary__body')
-
-  errorSummaryMessage = (): PageElement => cy.get('.govuk-error-message')
```
Comment (andrewrlee) [inline integration_tests/pages/bookedtoday/arrivals/imprisonmentStatus.ts:17]: moved all this into the super class and hid behind helper method

File: integration_tests/pages/page.ts [hunk 12]
```diff
@@ -16,4 +16,10 @@ export default abstract class Page {
   signOut = (): PageElement => cy.get('[data-qa=signOut]')
 
   manageDetails = (): PageElement => cy.get('[data-qa=manageDetails]')
+
+  hasError = (message: string) => {
+    cy.get('#error-summary-title').contains('There is a problem')
+    cy.get('.govuk-error-summary__body').contains(message)
+    cy.get('.govuk-error-message').contains(message)
+  }
```
Comment (GurnankCheema) [inline integration_tests/pages/page.ts:24]: nice touch!

**Pull request merged**